### ISSUE TITLE
CA1822 False Positive

### ISSUE THREAD

**Issue by PhilippNaused** [dotnet/roslyn-analyzers#7447]
### Analyzer

**Diagnostic ID**: [CA1822](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1822): `Mark members as static`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 9.0.100-rc.2.24474.11](https://dotnet.microsoft.com/download/dotnet/9.0)

_AND_

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 9.0.0-preview.24454.1

### Describe the bug

`CA1822` is reported on a partial method implementation that is implementing an interface.
Since it's is an interface implementation, "fixing" it causes [Compiler Error CS0736](https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0736).

### Steps To Reproduce

```cs
interface IFoo
{
    int Bar();
}

partial class Foo : IFoo
{
    public partial int Bar();
}

partial class Foo : IFoo
{
    public partial int Bar()
    {
        return 42;
    }
}
```

## Additional context

I noticed in the IDE that VS (17.12.0-pre.2.1) doesn't seem to recognize the implementation of `Bar()` as implementing an interface.
See the missing icon in line 16:
![Image](https://github.com/user-attachments/assets/01af93ef-d101-46dc-b211-624606524604)

**Comment by PhilippNaused** [comment]
Still reproducible in 10.0.100-rc.2.25502.107
